src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/writefinaldecision/WriteFinalDecisionMidEventHandler.java

```diff
@@ -60,4 +93,60 @@ private boolean isDecisionNoticeDateOfDecisionInvalid(SscsCaseData sscsCaseData)
         return false;
     }
 
+    @SuppressWarnings("squid:S1172")
+    @Override
+    protected DirectionOrDecisionIssuedTemplateBody createPayload(SscsCaseData caseData, String documentTypeLabel, LocalDate dateAdded, boolean isScottish,
+        String userAuthorisation) {
+        DirectionOrDecisionIssuedTemplateBody formPayload = super.createPayload(caseData, documentTypeLabel, dateAdded, isScottish, userAuthorisation);
+
+        DirectionOrDecisionIssuedTemplateBodyBuilder builder = formPayload.toBuilder();
+
+        builder.heldBefore(buildHeldBefore(caseData, userAuthorisation));
+
+        if (caseData.getHearings() != null && !caseData.getHearings().isEmpty()) {
+            Hearing finalHearing = caseData.getHearings().get(caseData.getHearings().size() - 1);
```

Comment:
Pretty sure hearings are sorted in latest date order when the callback is first deserialised (see here: https://github.com/hmcts/sscs-common/blob/master/src/main/java/uk/gov/hmcts/reform/sscs/ccd/domain/SscsCaseData.java#L531) so think you need to change this to `.get(0)`

Reply:
Good spot! - as discussed I've now changed this and added an integration test with multiple hearings which would have picked this up.